.github/workflows/build-and-deploy.yml

```diff
@@ -89,3 +89,61 @@ jobs:
         healthcheck: ${{ env.HEALTHCHECK_CMD }}
         db-seed: false
         smoke-test: false
+
+  deploy:
+    name: Deploy environments
+    if: ${{ github.ref == 'refs/heads/main' && github.event_name == 'push' }}
+    environment:
+      name: ${{ matrix.environment }}
+      url: ${{ steps.deploy_app.outputs.environment_url }}
+    needs: [build]
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      pull-requests: write
+    strategy:
+      max-parallel: 1
+      matrix:
+        environment: [staging, production]
```

Comment:
lets just build staging first. Prod in a separate PR please, once we have staging infra finished.